### ISSUE TITLE
Add support for keywords

### ISSUE THREAD

**Issue by justinethier** [justinethier/cyclone#423]
This is not a high priority ticket but would be a "nice to have" at some point in the future.

**Comment by justinethier** [comment]
Note this is SRFI 88 - https://srfi.schemers.org/srfi-88/srfi-88.html

It would probably make more sense as a core part of the compiler though.

**Comment by arthurmaciel** [comment]
Justin, coincidently I have studied this subject since issue #387. 

Related to SRFI-88 there is also https://srfi.schemers.org/srfi-89/srfi-89.html and https://srfi.schemers.org/srfi-177/srfi-177.html.

I have been thinking lately in a package proposal that handles this subject and maybe it would be useful to the core. It would make keywords unnecessary (as SRFI-177 syntax already does). I will try to present it here in the next weeks!

**Comment by arthurmaciel** [comment]
@justinethier, I was wrong. My implementation is orthogonal to keywords. Please see https://github.com/arthurmaciel/mu.

Best

**Comment by justinethier** [comment]
Thanks @arthurmaciel let me take a look at mu.

**Comment by arthurmaciel** [comment]
After thinking a lot about it, I am in favor of adding keywords support.

**Comment by arthurmaciel** [comment]
@justinethier, I far as I understand, keywords are mainly useful for allowing the call of procedures with named parameters, as [SRFI-89](https://srfi.schemers.org/srfi-89/srfi-89.html) proposes. So I suppose you intend to support it too, right?

Have you thought about making the keyword args definition less verbose?

Instead of defining an internal variable name `k` to represent `key:`
```Scheme 
(define (h1 a (key: k #f) . r)
  (list a k r))
```
wouldn't it be clearer to convert the keyword `key:` to a symbol `key`?
```Scheme 
(define (h1 a (key: #f) . r)
  (list a key r)) ;; no 'k', only 'key:' and 'key'
```
I don't know if that makes sense to you.

**Comment by justinethier** [comment]
@arthurmaciel Completely agree this would be a better way to make keyword arguments less verbose. Will have to look at SRFI 89 and other work in this area. I know with bigloo optional arguments the keyword is used directly in this same manner.

The initial goal here would be keyword support with optional args as a separate effort afterwards. But it does make sense to do that once we have keyword support.

Both of these are a bit on the horizon though. The shorter term goal is issue #193 (which is not directly related but also a larger task on our roadmap).

**Comment by arthurmaciel** [comment]
I completely agree with you that #193 is a top priority improvement. It is a shame I can't help with it.
